### PR TITLE
Remove tools from go.mod dependencies

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Build tools were removed from dependencies list in `go.mod` file. I will switch back to `go install` after go 1.16 release.

**Why?**
Build tools are not server dependencies and should not be listed in `go.mod`.

**How did you test it?**
Run locally (with empty cache) and on Buildkite.

**Potential risks**
No risks.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -124,21 +124,21 @@ GOCOVERPKG_ARG := -coverpkg="$(MODULE_ROOT)/common/...,$(MODULE_ROOT)/service/..
 ##### Tools #####
 update-checkers:
 	@printf $(COLOR) "Install/update check tools..."
-	go install golang.org/x/tools/cmd/goimports
-	go install golang.org/x/lint/golint
-	go install honnef.co/go/tools/cmd/staticcheck
-	go install github.com/kisielk/errcheck
-	go install github.com/googleapis/api-linter/cmd/api-linter
-	go install github.com/bufbuild/buf/cmd/buf
+	cd && GO111MODULE=on go get golang.org/x/tools/cmd/goimports
```
Comment (wxing1292) [inline Makefile:127]: why there is a `cd && `

File: go.sum [hunk 2]
```diff
@@ -1,7 +1,3 @@
-bitbucket.org/creachadair/stringset v0.0.8 h1:gQqe4vs8XWgMyijfyKE6K8o4TcyGGrRXe0JvHgx5H+M=
```
Comment (wxing1292) [inline go.sum:1]: why go sum is deleted?

**Comment by wxing1292** [comment]
for go.sum, i believe we should add it back
for the tools version, either make the tools as part of the dependency (tools.go) or go get tools/dependency@version

File: Makefile (hunk 1 above)
Comment (alexshtin) [reply Makefile:127]: I need to go to some dir which is outside of `$GOPATH` and doesn't have `go.mod`. `cd $(mktemp -d)` is another option but I don't want to pollute `/tmp` w/o good reason.

File: go.sum (hunk 2 above)
Comment (alexshtin) [reply go.sum:1]: It should be in sync with `go.mod`. I removed it from there and `go mod tidy` removed it from here.

**Review by wxing1292 — APPROVED** [review]

**Pull request merged**